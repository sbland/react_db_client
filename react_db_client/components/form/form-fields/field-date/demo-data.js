export const defaultVal = new Date(Date.now());