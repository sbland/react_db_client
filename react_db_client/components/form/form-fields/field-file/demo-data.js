export const DEMO_FILES_DATA = [
    { uid: 'a', filePath: 'files/fileA.jpg', name: 'FileA' },
    { uid: 'b', filePath: 'files/fileB.jpg', name: 'FileA' },
    { uid: 'c', filePath: 'files/fileC.jpg', name: 'FileA' },
  ];
