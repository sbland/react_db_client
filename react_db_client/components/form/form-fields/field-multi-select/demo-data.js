export const demoOptions = [
    {
        uid: 'a',
        label: 'A',
    },
    {
        uid: 'b',
        label: 'B',
    },
    {
        uid: 'c',
        label: 'C',
    },
];