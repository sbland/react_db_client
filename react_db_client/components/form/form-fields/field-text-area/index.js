export { FieldTextArea } from './field-text-area';
