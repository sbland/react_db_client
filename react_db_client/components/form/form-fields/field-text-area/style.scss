.inputWrapper {
    padding: 2px;
    margin: 0;
    outline: var(--default-field-outline, 1px solid #ddd);
}