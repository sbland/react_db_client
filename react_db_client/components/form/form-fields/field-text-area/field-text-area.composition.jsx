import React, { useState } from 'react';
import { CompositionWrapDefault, WrapFieldComponent } from '@samnbuk/react_db_client.helpers.composition-wraps';
import { FieldTextArea } from './field-text-area';
import { defaultVal } from './demo-data';

const updateFormData = () => {};

const defaultProps = {
  uid: 'uid',
  unit: 'unit',
  value: defaultVal,
  updateFormData,
};


export const BasicFieldTextArea = () => (
  <CompositionWrapDefault height="4rem" width="16rem">
    <FieldTextArea {...defaultProps} required />
  </CompositionWrapDefault>
);

export const FieldTextAreaMultiple = () => (
  <CompositionWrapDefault height="4rem" width="16rem">
    <FieldTextArea {...defaultProps} required />
    <FieldTextArea {...defaultProps} required />
    <FieldTextArea {...defaultProps} required />
  </CompositionWrapDefault>
);

export const FieldTextAreaMultipleHoriz = () => {
  return (
    <CompositionWrapDefault height="4rem" width="16rem">
      <div style={{ display: 'flex' }}>
        <WrapFieldComponent>
          <FieldTextArea {...defaultProps} required />
          <FieldTextArea {...defaultProps} required />
          <FieldTextArea {...defaultProps} required />
        </WrapFieldComponent>
      </div>
    </CompositionWrapDefault>
  );
};
