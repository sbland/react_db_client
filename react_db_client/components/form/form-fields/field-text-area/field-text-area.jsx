import React, { useState, useEffect } from 'react';
import PropTypes from 'prop-types';

import './style.scss';

export const FieldTextArea = ({
  uid,
  unit,
  updateFormData,
  value,
  required,
  initHeight,
  scaleToContent,
  styleOverrides,
}) => {
  const ref = React.useRef(null);
  const [textareaHeight, setTextareaHeight] = useState(initHeight);
  const [resizing, setResizing] = useState(scaleToContent);

  const manageInputChange = (newValue) => {
    // This makes the textbox auto size
    updateFormData(uid, newValue);
    if (scaleToContent) {
      setResizing(true);
    }
  };

  useEffect(() => {
    if (scaleToContent) {
      setResizing(true);
    }
  }, [value]);

  useEffect(() => {
    if (scaleToContent && resizing) {
      if (ref.current.scrollHeight + 2 > initHeight) {
        setTextareaHeight(ref.current.scrollHeight);
      }
      setResizing(false);
    }
  }, [scaleToContent, value, resizing, ref, initHeight]);

  return (
    <>
      <div
        className="inputWrapper"
        style={{
          ...styleOverrides,
        }}
      >
        <textarea
          style={{
            height: resizing ? 'auto' : textareaHeight + 2 + 'px',
            // height: textareaHeight + 'px',
            maxHeight: '100%',
            minHeight: '1rem',
            maxWidth: '100%',
            // minWidth: '100%',
            ...styleOverrides,
          }}
          value={value || ''}
          id={uid}
          name={uid}
          onFocus={() => {
            ref.current.select();
          }}
          onChange={(e) => manageInputChange(e.target.value)}
          required={required}
          ref={ref}
        />
      </div>
      {unit && <span>{unit}</span>}
    </>
  );
};

FieldTextArea.propTypes = {
  uid: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
  unit: PropTypes.string,
  value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
  updateFormData: PropTypes.func.isRequired,
  required: PropTypes.bool,
  initHeight: PropTypes.number,
  scaleToContent: PropTypes.bool,
  styleOverides: PropTypes.shape({}),
};

FieldTextArea.defaultProps = {
  unit: '',
  value: '',
  required: false,
  initHeight: 10,
  scaleToContent: true,
  styleOverides: {},
};
