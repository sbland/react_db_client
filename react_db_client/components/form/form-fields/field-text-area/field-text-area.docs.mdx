---
description: 'A React Form Field Component for editing long text.'
labels: ['form-field', 'field-text-area']
---

import { FieldTextArea } from './field-text-area';

