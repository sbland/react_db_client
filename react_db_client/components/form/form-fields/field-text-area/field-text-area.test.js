import '@samnbuk/react_db_client.helpers.enzyme-setup';
import React from 'react';
import { mount, shallow } from 'enzyme';

import { FieldTextArea } from "./field-text-area";
import * as compositions from './field-text-area.composition';
import { defaultVal } from './demo-data';

const updateFormData = jest.fn();

const defaultProps = {
  uid: 'uid',
  unit: 'unit',
  value: defaultVal,
  updateFormData,
};


describe('field-text-area', () => {
  beforeEach(() => {
    updateFormData.mockClear();
  });
  test('Renders', () => {
    shallow(<FieldTextArea {...defaultProps} />);
  });

  describe('Compositions', () => {
    Object.entries(compositions).forEach(([name, Composition]) => {
      test(name, () => {
        mount(<Composition />);
      });
    });
  });
  describe('shallow renders', () => {
    test('Matches Snapshot', () => {
      const component = shallow(<FieldTextArea {...defaultProps} />);
      const tree = component.debug();
      expect(tree).toMatchSnapshot();
    });
  });
});

