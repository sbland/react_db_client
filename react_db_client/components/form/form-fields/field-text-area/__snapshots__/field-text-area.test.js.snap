// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`field-text-area shallow renders Matches Snapshot 1`] = `
"<Fragment>
  <div className=\\"inputWrapper\\" style={{...}}>
    <textarea style={{...}} value=\\"In Informatics, dummy data is benign information that does not contain any useful data, but serves to reserve space where real data is nominally present.\\\\nDummy data can be used as a placeholder for both testing and operational purposes.\\\\n\\\\n\\\\nFor testing, dummy data can also be used as stubs or pad to avoid software testing issues by ensuring that all variables and data fields are occupied. In operational use, dummy data may be transmitted for OPSEC purposes. Dummy data must be rigorously evaluated and documented to ensure that it does not cause unintended effects.\\\\n\\\\n\\" id=\\"uid\\" name=\\"uid\\" onFocus={[Function: onFocus]} onChange={[Function: onChange]} required={false} />
  </div>
  <span>
    unit
  </span>
</Fragment>"
`;
